Add note that only file rules can be used to validate files (#7969)
codeigniter4/CodeIgniter4: user_guide_src/source/libraries/validation.rst

======================= ========== ============================================= ===================================================
uploaded                Yes         Fails if the name of the parameter does not  ``uploaded[field_name]``
                                    match the name of any uploaded files.
                                    Prevents form from validating if file upload
                                    is required.

> 

Reviewer: The description already provides the perfect explanation.

> The validation will fail if the name of the parameter does not match the name of any uploaded files.

Reviewer 2: I don't get what the added sentence means.

Reviewer 2: How about adding this?

> If you want the file upload to be optional (not required), do not define this rule.

Author: > How about adding this?
> 
> > If you want the file upload to be optional (not required), do not define this rule.

Yep, that is better, thanks.

Author: > The description already provides the perfect explanation.
> 
> > The validation will fail if the name of the parameter does not match the name of any uploaded files.

Technically you must be correct, but there must be a more user-friendly way to say "this rule makes the file required". I read that section and did not get it.

Reviewer: Yes. https://github.com/codeigniter4/CodeIgniter4/pull/7969#discussion_r1335243837 is better.

======================= ========== ============================================= ===================================================

The file validation rules apply for both single and multiple file uploads.

.. note:: Only rules specifically created for file validation (like the ones listed in the table above) can be used to validate files.

>     Therefore, adding any general rules, like ``permit_empy``, to file validation rules array or string, the file validation will not

Reviewer 2: ```suggestion
    Therefore, adding any general rules, like ``permit_empty``, to file validation rules array or string, the file validation will not
```

>     Therefore, adding any general rules, like ``permit_empy``, to file validation rules array or string will prevend all rules for the

Reviewer: A typo:
```suggestion
    Therefore, adding any general rules, like ``permit_empty``, to file validation rules array or string will prevend all rules for the
```

>     from file from being processed during validation.

Reviewer 2: I don't get well "prevend all rules for the from file from being processed during validation."

Reviewer 2: How about this?

```suggestion
    Therefore, if you add any general rules, like ``permit_empy`` to file validation rules array or string,
    the file validation will not work correctly.
```

Author: Well, if you set rules 'permit_empty|max_size[profile_img,1024]|mime_in[profile_img,image/jpeg]', you can uplood 3mb pdf's, and validation will pass.

Author: Yep, but I cannot believe I messed up my English that much... sorry.

Reviewer 2: ``permit_empty`` (or other general rules) checks `$_POST` array, and the field for the file upload does not exist, so the validation passes.
However, most rules do not permit empty, and `$_POST` has no field for it, so if set, the validation will fail.

I don't know how to explain it briefly.

Author: Ah, it's complicated. I was also thinking about filing a bug report (it is not good that adding a stray rule disables the other rules), but now I suspect that it is more of a limitation of validation than a bug...

>     Therefore, adding any general rules, like ``permit_empy``, to file validation rules array or string will prevend all rules for the
>     from file from being processed during validation.

Reviewer 2: How about this?

```suggestion
    Rules for General Use like ``permit_empy`` do not check the uploaded file (in the ``$_FILES`` superglobal),
    therefore, adding these rules to file validation rules array or string will not yield the expected results.
```
